.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,24 @@
+name: Lint Go Code
+
+on:
+  pull_request:
+    paths:
+      - '**/*.go'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
```

Comment:
also here, there is already v4 which is preferable: https://github.com/actions/setup-go